test/modules/TestResultCoordinator/CountingReportGenerator.cs

```diff
@@ -0,0 +1,207 @@
+            this.actualStore = Preconditions.CheckNotNull(actualStore, nameof(actualStore));
+            this.resultType = Preconditions.CheckNonWhiteSpace(resultType, nameof(resultType));
+            this.testResultComparer = Preconditions.CheckNotNull(testResultComparer, nameof(testResultComparer));
+            this.batchSize = batchSize;
+        }
+
+        internal string ActualSource => this.actualSource;
+
+        internal ISequentialStore<TestOperationResult> ActualStore => this.actualStore;
+
+        internal string ExpectedSource => this.expectedSource;
+
+        internal ISequentialStore<TestOperationResult> ExpectedStore => this.expectedStore;
+
+        internal string ResultType => this.resultType;
+
+        internal ITestResultComparer<TestOperationResult> TestResultComparer => this.testResultComparer;
+
+        internal int BatchSize => this.batchSize;
+
+        /// <summary>
+        /// Compare 2 data stores and counting expect, match, and duplicate results; and return a counting report.
+        /// It will remove consecutive duplicate results when loading from actual store.
+        /// It will log fail if actual store has more results than expect store.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<ITestResultReport> CreateReportAsync()
+        {
+            Logger.LogInformation($"Start to generate report by {nameof(CountingReportGenerator)} for Sources [{this.expectedSource}] and [{this.actualSource}]");
+
+            long lastLoadedKeyFromExpectStore = -1;
+            long lastLoadedKeyFromActualStore = -1;
+            TestOperationResult lastLoadedResult = default(TestOperationResult);
+
+            var expectQueue = new Queue<TestOperationResult>();
+            var actualQueue = new Queue<TestOperationResult>();
+            totalExpectCount = 0;
+            totalMatchCount = 0;
+            totalDuplicateResultCount = 0;
+            unmatchedResults = new List<TestOperationResult>();
+
+            (lastLoadedKeyFromExpectStore, _) = await this.LoadBatchIntoQueueAsync(this.expectedSource, this.expectedStore, lastLoadedKeyFromExpectStore, expectQueue);
+            (lastLoadedKeyFromActualStore, lastLoadedResult) = await this.LoadBatchIntoQueueAsync(this.actualSource, this.actualStore, lastLoadedKeyFromActualStore, actualQueue, lastLoadedResult);
+
+            while (expectQueue.Count > 0 && actualQueue.Count > 0)
+            {
+                TestOperationResult expectedResult = expectQueue.Dequeue();
+                TestOperationResult actualResult = actualQueue.Peek();
+
+                this.totalExpectCount++;
+
+                if (this.testResultComparer.Matches(expectedResult, actualResult))
+                {
+                    actualQueue.Dequeue();
+                    this.totalMatchCount++;
+                }
+                else
+                {
+                    this.unmatchedResults.Add(expectedResult);
```

Comment:
>unmatchedResults [](start = 25, length = 16)

If actual is: 1 1 2 3 4 and expected is 1 2 3 4, the second 1 gets reported as unmatched, but is actually a duplicate

Reply:
Already have test case covered; results loaded to actual queue already remove duplicates.